openassessment/workflow/models.py

```diff
@@ -279,23 +284,21 @@ def update_from_assessments(self, assessment_requirements, force_update_score=Fa
                 updates the problem definition.
 
         """
-        if self.status == self.STATUS.cancelled:
-            return
-
         # Update our AssessmentWorkflowStep models with the latest from our APIs
         steps = self._get_steps()
 
         step_for_name = {step.name: step for step in steps}
 
-        # If the status is done or cancelled, check if score has changed.
-        if self.status == self.STATUS.done:
-            if force_update_score:
-                new_score = self.get_score(assessment_requirements, step_for_name)
-                self.set_score(new_score)
-                self.save()
-                logger.info((
-                    u"Workflow for submission UUID {uuid} has updated score."
-                ).format(uuid=self.submission_uuid))
+        if force_update_score:
+            new_score = self.get_score(assessment_requirements, step_for_name)
+            self.set_staff_score(new_score, is_override=True)
+            self.save()
+            logger.info((
+                u"Workflow for submission UUID {uuid} has updated score."
+            ).format(uuid=self.submission_uuid))
+            return
+
+        if self.status == self.STATUS.cancelled or self.status == self.STATUS.done:
```

Comment:
I had meant to ask this earlier - are we going to allow staff overrides on cancelled workflows, or should I keep the previous behavior of checking for `STATUS.cancelled` before handling `force_update_score`?